### PR TITLE
IPFS support for large files in git remote

### PR DESCRIPTION
**Pull request by awnion**

### CODE REVIEW

File: git-remote-gosh/ipfs.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+ *      curl -X POST -H "Content-Type: application/json" \
+ *        -d '{"cid": "QmUtV6dzFDS6JNkj2e2dp5vxQPogdF58zQYwRL5v1LGEQe"}' \
+ *        http://pin.ipfs.gosh.sh:9097/pinss
+ * @param {String} cid 
+ * @link https://ipfs.github.io/pinning-services-api-spec/#tag/pins/paths/~1pins~1{requestid}/get
+ */
+async function pinRemoteAdd(cid) {
+    let response
+    while (networkErrors < MAX_NETWORK_ERRORS) {
+        console.log(`${networkErrors} POST pin ${cid}`)
```
Comment (sauin) [inline git-remote-gosh/ipfs.js:43]: `console.log` sends data to stdout which will be intercepted and sent to the git client. use `console.error` (sends to stderr) or better use verbose()

File: git-remote-gosh/everscale.js [hunk 2]
```diff
@@ -19,6 +20,20 @@ let CURRENT_REPO
 let Gosh, Repository, Commit, Blob, Dao, Tag
 let UserWallet = {}
 
+const MAX_ONCHAIN_FILE_SIZE = 16384;
+
+const gitOpCosts = {
```
Comment (sauin) [inline git-remote-gosh/everscale.js:25]: deprecated

File: git-remote-gosh/everscale.js [hunk 3]
```diff
@@ -19,6 +20,20 @@ let CURRENT_REPO
...
+    createRepo: 2.7e9,
+    createBranch: 0.22e9,
+    deleteBranch: 0.11e9,
+}
+
+const verbose = (...data) => console.error(...data)
+const fatal = (...data) => {
```
Comment (sauin) [inline git-remote-gosh/everscale.js:32]: `verbose()` and `fatal()` already imported (see above): `const { verbose, fatal } = require('./utils')`

**Review by sauin — CHANGES_REQUESTED** [review]

File: git-remote-gosh/everscale.js (hunk 2 above)
Comment (awnion) [reply git-remote-gosh/everscale.js:25]: yeap merge bug

File: git-remote-gosh/everscale.js (hunk 3 above)
Comment (awnion) [reply git-remote-gosh/everscale.js:32]: same, merge bug

File: git-remote-gosh/ipfs.js (hunk 1 above)
Comment (awnion) [reply git-remote-gosh/ipfs.js]: fixed

File: git-remote-gosh/everscale.js (hunk 2 above)
Comment (awnion) [reply git-remote-gosh/everscale.js]: fixed

File: git-remote-gosh/everscale.js (hunk 3 above)
Comment (awnion) [reply git-remote-gosh/everscale.js]: fixed

**Review by sauin — APPROVED** [review]

**Pull request merged**